includes/admin/home/tasks/task/class-sensei-home-task-pro-upsell.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * File containing the Sensei_Home_Task_Pro_Upsell class.
+ *
+ * @package sensei-lms
+ * @since $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Home_Task_Pro_Upsell.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_Home_Task_Pro_Upsell implements Sensei_Home_Task {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'sensei_home_tasks', [ $this, 'maybe_add_external_icon_property' ] );
+	}
+
+	/**
+	 * Maybe add external icon to the task.
+	 *
+	 * @param array $tasks Tasks.
+	 *
+	 * @return array
+	 */
+	public function maybe_add_external_icon_property( $tasks ) {
+		return array_map(
+			function ( $task ): array {
+				if ( self::get_id() === $task['id'] ) {
+						$task['external_icon'] = true;
+				}
+				return $task;
+			},
+			$tasks
+		);
+	}
+
+
+	/**
+	 * The ID for the task.
+	 *
+	 * @return string
+	 */
+	public static function get_id(): string {
+		return 'sensei-home-task-pro-upsell';
+	}
+
+	/**
+	 * Number used to sort in frontend.
+	 *
+	 * @return int
+	 */
+	public function get_priority(): int {
+		return 250;
+	}
+
+	/**
+	 * Task title.
+	 *
+	 * @return string
+	 */
+	public function get_title(): string {
+		return __( 'Sell your course with Sensei Pro', 'sensei-lms' );
+	}
+
+	/**
+	 * Task url.
+	 *
+	 * @return string
+	 */
+	public function get_url(): ?string {
+		return rest_url( 'sensei-internal/v1/home/sensei-pro-upsell-redirect?_wpnonce=' . wp_create_nonce( 'wp_rest' ) );
+	}
+
+	/**
+	 * Whether the task is completed or not.
+	 *
+	 * @return bool
+	 */
+	public function is_completed(): bool {
+		return get_option( self::get_id(), false );
+	}
+
+	/**
+	 * Mark the task as completed.
+	 *
+	 * @internal
+	 */
+	public static function mark_completed_and_redirect() {
+		sensei_log_event( 'home_task_complete', [ 'type' => self::get_id() ] );
+
+		update_option( self::get_id(), true );
+
+		// phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect -- We're redirecting to an external URL.
+		wp_redirect( 'https://senseilms.com/sensei-pro/?utm_source=plugin_sensei&utm_medium=upsell&utm_campaign=sensei-home' );
+	}
+
+	/**
+	 * Whether the task is active or not.
+	 *
+	 * @return bool
+	 */
+	public static function is_active(): bool {
+		if ( ! function_exists( 'get_plugins' ) ) {
```

Comment:
And I think the same thing as https://github.com/Automattic/sensei-pro/pull/2566#issuecomment-2023493822 would apply here, about the selected features.

Reply:
Makes sense! I've updated it here https://github.com/Automattic/sensei/pull/7562/commits/db79c14b33eebb8b28ee49dcea5c3930f6c4c9a6

Thanks!